src/ORM/Association/HasMany.php

```diff
@@ -104,6 +147,11 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
         $original = $targetEntities;
         $options['_sourceTable'] = $this->source();
 
+        if ($this->_saveStrategy === self::SAVE_REPLACE) {
+            $updateFields = array_fill_keys(array_keys($properties), null);
```

Comment:
I think the action taken here needs to be different depending on the `dependent` configuration option and also the `nullable` flag for the foreign key. In some cases, this needs to be a cascading delete.